verifier/rest/rest.go

```diff
@@ -6,40 +6,58 @@ import (
 	"encoding/base64"
 	"fmt"
 	"log"
-	"strings"
+	"time"
 
+	"github.com/cenkalti/backoff/v4"
 	sabi "github.com/google/go-sev-guest/abi"
 	"github.com/google/go-sev-guest/proto/sevsnp"
 	tabi "github.com/google/go-tdx-guest/abi"
 	"github.com/google/go-tdx-guest/proto/tdx"
 	"github.com/google/go-tpm-tools/verifier"
 	"github.com/google/go-tpm-tools/verifier/oci"
+	"github.com/googleapis/gax-go/v2"
 
 	v1 "cloud.google.com/go/confidentialcomputing/apiv1"
 	confidentialcomputingpb "cloud.google.com/go/confidentialcomputing/apiv1/confidentialcomputingpb"
 	"google.golang.org/api/iterator"
 	"google.golang.org/api/option"
 	locationpb "google.golang.org/genproto/googleapis/cloud/location"
+	"google.golang.org/grpc/codes"
 )
 
-// BadRegionError indicates that:
-//   - the requested Region cannot be used with this API
-//   - other Regions _can_ be used with this API
-type BadRegionError struct {
-	RequestedRegion  string
-	AvailableRegions []string
-	err              error
+func confComputeBackoffPolicy() backoff.BackOff {
+	b := backoff.NewConstantBackOff(time.Millisecond * 500)
+	return backoff.WithMaxRetries(b, 3)
 }
 
-func (e *BadRegionError) Error() string {
-	return fmt.Sprintf(
-		"invalid region %q, available regions are [%s]: %v",
-		e.RequestedRegion, strings.Join(e.AvailableRegions, ", "), e.err,
-	)
-}
-
-func (e *BadRegionError) Unwrap() error {
-	return e.err
+/*
+confComputeCallOptions retries as follows for all confidential computing APIs:
+
+	Timeout = 1000 milliseconds
+	Initial interval = 500 milliseconds
+	Maximum interval = 1000 milliseconds
+	Maximum retries = 2
+*/
+func confComputeCallOptions() *v1.CallOptions {
+	callOption := []gax.CallOption{
+		gax.WithTimeout(1000 * time.Millisecond),
+		gax.WithRetry(func() gax.Retryer {
+			return gax.OnCodes([]codes.Code{
+				codes.Unavailable,
+				codes.Internal,
```

Comment:
Seems like your retry option only going to retry on these two codes. Do we need to worry about other codes (like in here: https://cloud.google.com/storage/docs/retry-strategy#go)

Reply:
These codes are actually the mapping of gRPC error codes in https://github.com/grpc/grpc/blob/master/doc/statuscodes.md. And yes, these are only two error codes we want to retry on because 1. code.Unavailable is the default error code to retry 2. we only observed 500 internal error codes so far which caused test flakiness.